src/ui/web/src/components/charging-stations/CSConnector.vue

```diff
@@ -61,9 +61,9 @@ function hideTagModal(): void {
 }
 
 function startTransaction(): void {
-  UIServer.startTransaction(props.hashId, props.connectorId, state.tag);
+  UIClient.instance.startTransaction(props.hashId, props.connectorId, state.tag);
```

Comment:
The reason i made all method static was to not have the redundancy of instance. Is it not "cleaner" the old way ?

Reply:
Either you use a class with static methods only, fordid instantiation, static attributes to ensure that they will be always put in memory once, either you implement a singleton class but not both. 
Looks at your original code: 
+ indirection with setters to get the running instance safely (useless branching) in hot code path
+ indirection with setters to access the class attributes safely: the whole purpose of OOP is to do class that permits to access its attributes directly, without overhead and safely per design. 

The singleton design pattern UML classes diagram: https://en.wikipedia.org/wiki/Singleton_pattern

Reply:
I understand the problem with branching, didn't think of it.